src/jit/codegenxarch.cpp

```diff
@@ -8172,9 +7955,11 @@ CodeGen::genPutStructArgStk(GenTreePtr treeNode, unsigned baseVarNum)
         // They may now contain gc pointers (depending on their type; gcMarkRegPtrVal will "do the right thing").
         genConsumePutStructArgStk(putArgStk, REG_RDI, REG_RSI, REG_NA, baseVarNum);
         GenTreePtr   dstAddr = putArgStk;
-        GenTreePtr   srcAddr = putArgStk->gtOp.gtOp1;
+        GenTreePtr   src = putArgStk->gtOp.gtOp1;
```

Comment:
These code changes contain fixes to 3 different gc-hole issues

1) Issue of TypedReference not classfiied and handled in codegen
2) genPutStructArgStk() not using srcAddr type while emitting code
3) genConsumePutStructArgStk() not using srcAddr type while emitting code

Note that Issue#1 affects only mfrefany tests (which are 4 or 5 jit tests AFAIK) and this also touches classification code and as part of that there is a need to eliminate code duplication by restructuring code.

But Fixes to Issue#2 and #3 are orthogonal to issue#1 and could be checked-in separately and impacts lot of jit tests or real world code.  This is not only better for tracking purpose (e.g. rollback due to regressions etc) but also will unblock further gc-stress testing.